### PR TITLE
stylo: Store location information for keyframe rules.

### PR DESCRIPTION
**Pull request by chenpighead**
So far, we only store location info for the whole Keyframes block, not for each
of the keyframe rule. Without this info, the devtool can't present the rules
on the devtool panel properly.

In this patch, we collect the source location info while parsing keyframe selector.
The binding function, Servo_KeyframesRule_GetKeyframe, is also fixed (and renamed
to Servo_KeyframesRule_GetKeyframeAt to match the fix) to accept line and column
parameters from Gecko, so we can pass/set them with the source location from Servo.

This is the servo part of [Bug 1394994](https://bugzilla.mozilla.org/show_bug.cgi?id=1394994).

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix [Bug 1394994](https://bugzilla.mozilla.org/show_bug.cgi?id=1394994)

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/18365)

### CODE REVIEW

**Comment by chenpighead** [comment]
This has been reviewed by xidorn in [Bug 1394994.](https://bugzilla.mozilla.org/show_bug.cgi?id=1394994)

Gecko try: https://treeherder.mozilla.org/#/jobs?repo=try&revision=9a189267f4d06a8afffa04a81a79f915d3b05c21

**Comment by chenpighead** [comment]
Oops, lack of bindings.rs part I guess, I'll update the patch

**Comment by chenpighead** [comment]
Sorry, it's the unit test that I forgot to update... :(

File: tests/unit/style/stylesheets.rs [hunk 1]
```diff
@@ -217,7 +217,11 @@ fn test_parse_stylesheet() {
                                 (PropertyDeclaration::Width(
                                     LengthOrPercentageOrAuto::Percentage(Percentage(0.))),
                                  Importance::Normal),
-                            ])))
+                            ]))),
+                            source_location: SourceLocation {
+                                                 line: 17,
```
Comment (emilio) [inline tests/unit/style/stylesheets.rs]: nit: Align this four spaces to the right of the "s" in `source_location`

File: tests/unit/style/stylesheets.rs [hunk 2]
```diff
@@ -231,6 +235,10 @@ fn test_parse_stylesheet() {
                                         vec![TimingFunction::ease()])),
                                  Importance::Normal),
                             ]))),
+                            source_location: SourceLocation {
+                                                 line: 18,
```
Comment (emilio) [inline tests/unit/style/stylesheets.rs]: nit: And here too.

**Review by emilio — APPROVED** [review]
Unit test changes lgtm.

File: tests/unit/style/stylesheets.rs (hunk 2 above)
Comment (chenpighead) [reply tests/unit/style/stylesheets.rs]: No problem. Thanks for the quick review. :)

**Comment by chenpighead** [comment]
wat?
@bors-servo retry

**Comment by chenpighead** [comment]
seriously?
@bors-servo retry

**Comment by upsuper** [comment]
You would fail again with mac-dev-unit. The slave machine building this is somehow broken.

**Comment by chenpighead** [comment]
Hmm, I saw a PR that retry couple times and then get passed yesterday, thought I could get some lucky as well... maybe I'm running into a different issue...  :disappointed:

**Comment by upsuper** [comment]
That machine seems to really like your PR.

**Comment by chenpighead** [comment]
Filed #18374 for the failure...

**Pull request merged**